### PR TITLE
refactor: GridForm ColorModes + system props 

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview
Updates some final GridForm / Form components to match current ColorMode conventions.

Checkbox needed to be entirely refactored + I split into separate PR [here](https://github.com/Codecademy/client-modules/pull/1886)

### PR Checklist

- [x] Related to designs: 
- [x] Related to JIRA ticket: [GM-210](https://codecademy.atlassian.net/browse/GM-210)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.63-alpha.17a455.0<br/>@codecademy/gamut-labs@19.16.1-alpha.17a455.0<br/>@codecademy/gamut@34.14.2-alpha.17a455.0<br/>@codecademy/styleguide@41.14.1-alpha.17a455.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ec701bdd0de63db8587609--gamut-preview.netlify.app>https://60ec701bdd0de63db8587609--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ec701bdd0de63db8587609>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.63-alpha.6c5c4f.0<br/>@codecademy/gamut-labs@19.16.1-alpha.6c5c4f.0<br/>@codecademy/gamut@34.14.2-alpha.6c5c4f.0<br/>@codecademy/styleguide@41.14.1-alpha.6c5c4f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60edc528c125e817a1f82443--gamut-preview.netlify.app>https://60edc528c125e817a1f82443--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60edc528c125e817a1f82443>Deploy Logs</a></p>

File: packages/gamut/src/Form/FormGroupLabel.tsx [hunk 1]
```diff
@@ -48,29 +47,22 @@ const labelSizeVariants = variant({
...
-    return css`
-      color: ${theme.colors['gray-600']};
-    `;
-  }
-};
+const labelColorStates = states({
+  disabled: {
+    ...formFieldTextDisabledStyles,
+  },
+});
```
Comment (codecaaron) [inline packages/gamut/src/Form/FormGroupLabel.tsx:54]: Fixed the mutability issue here so this should be safe again :) 

```suggestion
const labelColorStates = states({
  disabled: formFieldTextDisabledStyles,
});
```

File: packages/gamut/src/Form/FormGroupLabel.tsx [hunk 2]
```diff
@@ -28,13 +30,10 @@ export type FormGroupLabelProps = HTMLAttributes<HTMLDivElement> &
...
   };
 
-type disabledLabelStyleProps = {
-  disabled?: boolean;
-};
-
 const labelSizeVariants = variant({
   defaultVariant: 'small',
   prop: 'size',
+  ...formBaseStyles,
```
Comment (codecaaron) [inline packages/gamut/src/Form/FormGroupLabel.tsx:36]: Shared styles need to be on the base key :) 
```suggestion
   base: formBaseStyles,
```

File: packages/gamut/src/Form/FormGroupLabel.tsx [hunk 3]
```diff
@@ -48,29 +47,22 @@ const labelSizeVariants = variant({
...
+});
 
-const formLabelStyles = ({ size, disabled }: FormGroupLabelProps) => css`
-  ${formBaseStyles}
-  ${disabledLabelStyle({ disabled })}
-  ${labelSizeVariants({ size })}
+const Label = styled.label`
+  ${labelSizeVariants}
+  ${labelColorStates}
   display: block;
```
Comment (codecaaron) [inline packages/gamut/src/Form/FormGroupLabel.tsx:59]: You could probably add this to `base: { display: 'block', ...formBaseStyles }` on https://github.com/Codecademy/client-modules/pull/1891/files#diff-70c18eed674e4c336d4e47633e799c6e1bbdbac3b6b7a781f555180273a13625R36

And you'd be able to do:
```tsx
const Label = styled.label(
  labelSizeVariants,
  labelColorStates
).withComponent()
```

File: packages/gamut/src/Form/Select.tsx [hunk 4]
```diff
@@ -107,9 +107,9 @@ export const Select = forwardRef<HTMLSelectElement, SelectWrapperProps>(
           aria-hidden
         >
           {sizeVariant === 'small' ? (
-            <MiniChevronDownIcon size={12} />
+            <MiniChevronDownIcon size={12} color="text" />
```
Comment (codecaaron) [inline packages/gamut/src/Form/Select.tsx:110]: Side note - for https://github.com/Codecademy/client-modules/pull/1891/files#diff-e8e618f0fa471d94e1b31b91051dc1140e91094189470c38911ea6c7f2b28fc0R100

I believe the `no-select` prop on FlexBox will work here :)

File: packages/gamut/src/Form/styles/shared-system-props.tsx [hunk 5]
```diff
@@ -26,26 +26,35 @@ export enum InputSelectors {
...
+  color: 'text-disabled',
+  cursor: 'not-allowed',
+} as const;
+
 const formFieldBaseDisabledStyles = {
   borderColor: 'currentColor',
-  textColor: 'text-disabled',
   fontStyle: 'italic',
-  cursor: 'not-allowed',
+  opacity: '1',
```
Comment (codecaaron) [inline packages/gamut/src/Form/styles/shared-system-props.tsx:56]: Just curious what we're doing with the opacity here?

File: packages/styleguide/stories/Organisms/examples.tsx [hunk 6]
```diff
@@ -0,0 +1,262 @@
+/* eslint-disable local-rules/gamut-import-paths */
+import { Box, GridForm } from '@codecademy/gamut/src';
+import React from 'react';
+
+import { ColorModeExampleWrapper } from '../Foundations/ColorMode/examples';
+
+export const ColorModeExample = () => {
```
Comment (codecaaron) [inline packages/styleguide/stories/Organisms/examples.tsx:7]: Love this!

Could we add a few more things to this?
- Errors
- Checkboxes (even iff they are broken lol)
- Tooltips?

File: packages/styleguide/stories/Organisms/examples.tsx [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+          ]}
+          onSubmit={() => {}}
+          submit={{
+            type: 'cta',
+            contents: 'CTA Button',
+            size: 12,
+          }}
+          validation="onSubmit"
+        />
+      </Box>
```
Comment (codecaaron) [inline packages/styleguide/stories/Organisms/examples.tsx:259]: Do we need this one?

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx [hunk 8]
```diff
@@ -987,3 +989,9 @@ We call it a "sweet container" so that bots do not immediately detect it as a ho
     )}
   </Story>
 </Canvas>
+
+## `<ColorMode />`
+
+Here's an example using components in context. By toggling dark mode you can see all the colors map to a new color that is accessible for the mode by default.
+
+<ColorModeExample />
```
Comment (codecaaron) [inline packages/styleguide/stories/Organisms/GridForm.stories.mdx:997]: Can we make a note about not using Submit buttons that aren't new gamut if you want this behavior?  Lots of button deprecated usages still in the wild.

File: packages/styleguide/stories/Foundations/ColorMode/examples.tsx [hunk 9]
```diff
@@ -84,7 +84,38 @@ const renderLinks = () => {
...
+      </Text>
+      {renderLinks()}
+      {renderButtons('primary', false, true)}
+      {renderButtons('primary')}
+      {renderButtons('primary', true)}
+      {renderButtons('secondary')}
+      {renderButtons('danger')}
+    </>
+  );
+};
```
Comment (codecaaron) [inline packages/styleguide/stories/Foundations/ColorMode/examples.tsx:116]: Would love if we had some input examples here too :D - thanks for cleaning this up!!

**Review by codecaaron — COMMENTED** [review]
Looking great!  Few suggestions / questions.

File: packages/gamut/src/Form/FormGroupLabel.tsx (hunk 1 above)
Comment (dreamwasp) [reply packages/gamut/src/Form/FormGroupLabel.tsx:54]: 🙏

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.63-alpha.f6c697.0<br/>@codecademy/gamut-labs@19.16.1-alpha.f6c697.0<br/>@codecademy/gamut@34.14.2-alpha.f6c697.0<br/>@codecademy/styleguide@41.14.1-alpha.f6c697.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60edd3f33c963534800f0f16--gamut-preview.netlify.app>https://60edd3f33c963534800f0f16--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60edd3f33c963534800f0f16>Deploy Logs</a></p>

File: packages/gamut/src/Form/styles/shared-system-props.tsx (hunk 5 above)
Comment (dreamwasp) [reply packages/gamut/src/Form/styles/shared-system-props.tsx:56]: it overrides the user agent stylesheet (which in this case is specific to `select` -- disabled styles are .7 at default, but wanted to future-proof a bit)

File: packages/styleguide/stories/Organisms/examples.tsx (hunk 6 above)
Comment (dreamwasp) [reply packages/styleguide/stories/Organisms/examples.tsx:7]: i have checkbox added in my branch off of this [one](https://github.com/Codecademy/client-modules/pull/1886?notification_referrer_id=MDE4Ok5vdGlmaWNhdGlvblRocmVhZDIxMjQ5MTQ2ODc6NTgwNTQ3NTE%3D#issuecomment-878582352) so will be merged in imminently, will add errors and tooltips soon!

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx (hunk 8 above)
Comment (dreamwasp) [reply packages/styleguide/stories/Organisms/GridForm.stories.mdx:997]: will do!

File: packages/gamut/src/Form/styles/shared-system-props.tsx (hunk 5 above)
Comment (codecaaron) [reply packages/gamut/src/Form/styles/shared-system-props.tsx:56]: 💯 TIL :D

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.65-alpha.ba339b.0<br/>@codecademy/gamut-labs@19.16.3-alpha.ba339b.0<br/>@codecademy/gamut@34.14.4-alpha.ba339b.0<br/>@codecademy/styleguide@41.14.5-alpha.ba339b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f1ea767e6e2f1262a29345--gamut-preview.netlify.app>https://60f1ea767e6e2f1262a29345--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f1ea767e6e2f1262a29345>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.66-alpha.97c7c4.0<br/>@codecademy/gamut-labs@19.16.4-alpha.97c7c4.0<br/>@codecademy/gamut@34.14.5-alpha.97c7c4.0<br/>@codecademy/styleguide@41.14.6-alpha.97c7c4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f1fdcd0571da2832c0a929--gamut-preview.netlify.app>https://60f1fdcd0571da2832c0a929--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f1fdcd0571da2832c0a929>Deploy Logs</a></p>

File: packages/gamut/src/Form/Select.tsx [hunk 10]
```diff
@@ -107,9 +115,9 @@ export const Select = forwardRef<HTMLSelectElement, SelectWrapperProps>(
           aria-hidden
         >
           {sizeVariant === 'small' ? (
-            <MiniChevronDownIcon size={12} />
+            <MiniChevronDownIcon size={12} color="currentColor" />
           ) : (
-            <ArrowChevronDownIcon size={16} />
+            <ArrowChevronDownIcon size={16} color="currentColor" />
```
Comment (codecaaron) [inline packages/gamut/src/Form/Select.tsx:120]: Hmm icons should be currentColor by default :/

File: packages/gamut/src/Form/FormGroupDescription.tsx [hunk 11]
```diff
@@ -1,15 +1,19 @@
...
-import styles from './styles/FormGroupDescription.module.scss';
+import { formBaseStyles } from './styles/shared-system-props';
 
 export type FormGroupDescriptionProps = React.HTMLAttributes<HTMLDivElement> & {
   className?: string;
 };
 
+const StyledDiv = styled.div<FormGroupDescriptionProps>`
+  ${system.css(formBaseStyles)}
+`;
```
Comment (codecaaron) [inline packages/gamut/src/Form/FormGroupDescription.tsx:13]: Do we even need the extra component now?

```suggestion
export const FormGroupDescription = styled.div(system.css(formBaseStyles));
```

**Review by codecaaron — APPROVED** [review]
💯 Awwwwesome

One small ⛳  shot for the eagle.

File: packages/gamut/src/Form/FormGroupDescription.tsx (hunk 11 above)
Comment (dreamwasp) [reply packages/gamut/src/Form/FormGroupDescription.tsx:13]: 🙏

Comment (dreamwasp) [reply packages/gamut/src/Form/FormGroupDescription.tsx]: we may still need to support className but i'll see if it's actually used anywhere...

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.66-alpha.781983.0<br/>@codecademy/gamut-labs@19.16.4-alpha.781983.0<br/>@codecademy/gamut@34.14.5-alpha.781983.0<br/>@codecademy/styleguide@41.14.6-alpha.781983.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f20314d73ef700dcd83f6e--gamut-preview.netlify.app>https://60f20314d73ef700dcd83f6e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f20314d73ef700dcd83f6e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.67-alpha.6ef08a.0<br/>@codecademy/gamut-labs@19.16.5-alpha.6ef08a.0<br/>@codecademy/gamut@34.15.1-alpha.6ef08a.0<br/>@codecademy/styleguide@41.15.1-alpha.6ef08a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f2099af706db09931b6a90--gamut-preview.netlify.app>https://60f2099af706db09931b6a90--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f2099af706db09931b6a90>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.69-alpha.626568.0<br/>@codecademy/gamut-labs@19.16.7-alpha.626568.0<br/>@codecademy/gamut@35.0.2-alpha.626568.0<br/>@codecademy/styleguide@42.0.2-alpha.626568.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f5d53b0016ae092b0ce30f--gamut-preview.netlify.app>https://60f5d53b0016ae092b0ce30f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f5d53b0016ae092b0ce30f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.70-alpha.df14e8.0<br/>@codecademy/gamut-labs@19.16.8-alpha.df14e8.0<br/>@codecademy/gamut@35.1.1-alpha.df14e8.0<br/>@codecademy/styleguide@42.1.1-alpha.df14e8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f6d6bb889d2400ab2b16c2--gamut-preview.netlify.app>https://60f6d6bb889d2400ab2b16c2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f6d6bb889d2400ab2b16c2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.70-alpha.d1df1e.0<br/>@codecademy/gamut-labs@19.16.8-alpha.d1df1e.0<br/>@codecademy/gamut@35.1.1-alpha.d1df1e.0<br/>@codecademy/styleguide@42.1.1-alpha.d1df1e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f8990a57a9fd4db8b74794--gamut-preview.netlify.app>https://60f8990a57a9fd4db8b74794--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f8990a57a9fd4db8b74794>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.70-alpha.9ce2d8.0<br/>@codecademy/gamut-labs@19.16.8-alpha.9ce2d8.0<br/>@codecademy/gamut@35.1.1-alpha.9ce2d8.0<br/>@codecademy/styleguide@42.1.1-alpha.9ce2d8.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.70-alpha.3b1240.0<br/>@codecademy/gamut-labs@19.16.8-alpha.3b1240.0<br/>@codecademy/gamut@35.1.1-alpha.3b1240.0<br/>@codecademy/styleguide@42.1.1-alpha.3b1240.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f8bc812202da0b12a58e8a--gamut-preview.netlify.app>https://60f8bc812202da0b12a58e8a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f8bc812202da0b12a58e8a>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.71-alpha.48f945.0<br/>@codecademy/gamut-labs@19.16.9-alpha.48f945.0<br/>@codecademy/gamut@35.1.2-alpha.48f945.0<br/>@codecademy/styleguide@42.1.2-alpha.48f945.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f8beb6067e720c3b5405c1--gamut-preview.netlify.app>https://60f8beb6067e720c3b5405c1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f8beb6067e720c3b5405c1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.71-alpha.6bdd20.0<br/>@codecademy/gamut-labs@19.16.9-alpha.6bdd20.0<br/>@codecademy/gamut@35.1.2-alpha.6bdd20.0<br/>@codecademy/styleguide@42.1.2-alpha.6bdd20.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60f98a3db6e9530dcfc1a364--gamut-preview.netlify.app>https://60f98a3db6e9530dcfc1a364--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60f98a3db6e9530dcfc1a364>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.71-alpha.bb4aef.0<br/>@codecademy/gamut-labs@19.16.9-alpha.bb4aef.0<br/>@codecademy/gamut@36.0.0-alpha.bb4aef.0<br/>@codecademy/styleguide@43.0.0-alpha.bb4aef.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ff2237470ba7534719af95--gamut-preview.netlify.app>https://60ff2237470ba7534719af95--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ff2237470ba7534719af95>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.74-alpha.e410c9.0<br/>@codecademy/gamut-labs@19.16.12-alpha.e410c9.0<br/>@codecademy/gamut@36.0.0-alpha.e410c9.0<br/>@codecademy/styleguide@43.0.0-alpha.e410c9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ff24623c1ac400b619d57e--gamut-preview.netlify.app>https://60ff24623c1ac400b619d57e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ff24623c1ac400b619d57e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.1-alpha.4cbf61.0<br/>@codecademy/gamut-labs@19.16.13-alpha.4cbf61.0<br/>@codecademy/gamut@36.0.0-alpha.4cbf61.0<br/>@codecademy/styleguide@43.0.0-alpha.4cbf61.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61000db6f3b85010ef030504--gamut-preview.netlify.app>https://61000db6f3b85010ef030504--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61000db6f3b85010ef030504>Deploy Logs</a></p>

**Pull request merged**